code/src/layouts/modals/auth/components/sign-in-form/SignInForm.tsx

```diff
@@ -0,0 +1,84 @@
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+
+import AppInput from "@/components/app-input/AppInput";
+import AppBox from "@/components/app-box/AppBox";
+import AppContainer from "@/components/app-container/AppContainer";
+import AppButton from "@/components/app-button/AppButton";
+import AppTypography from "@/components/app-typography/AppTypography";
+
+import { useModalContext } from "@/context/ModalContext";
+import useInputVisibility from "@/hooks/use-input-visibility/useInputVisibility";
+import useSignIn from "@/hooks/use-sign-in/useSignIn";
+import {
+  SignInVallidationScheme,
+  SignInVallidatorType
+} from "@/utils/validators/signInScheme";
+
+import "@/layouts/modals/auth/components/sign-in-form/SignInForm.scss";
+
+const SignInForm = () => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors }
+  } = useForm<SignInVallidatorType>({
+    resolver: zodResolver(SignInVallidationScheme)
+  });
+
+  const [signIn, { isLoading }] = useSignIn();
+
+  const { closeModal } = useModalContext();
+
+  const {
+    inputVisibility: passwordVisibility,
+    shouldShowInputText: showPassword
+  } = useInputVisibility();
+
+  const onSubmit = async ({ email, password }: SignInVallidatorType) => {
+    await signIn({ email, password });
+    closeModal();
+  };
```

Comment:
This logic does not work correct because `closeModal` should close only when there is successful sign in or sign up. Same issue for sign up

https://github.com/idx-academy/spa-orders/assets/109922101/0300f0a1-aa48-4565-b894-35d29375548b

Maybe we can use `then` here

Reply:
fixed